packer/conf/buildkite-agent/scripts/mark-asg-unhealthy

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -euo pipefail
+
+instance_id=$(curl -fsSL http://169.254.169.254/latest/meta-data/instance-id)
+region=$(curl -fsSL http://169.254.169.254/latest/meta-data/placement/availability-zone | head -c -1)
+
+aws autoscaling set-instance-health --instance-id "$instance_id" --region "$region" --health-status "Unhealthy"
```

Comment:
Would love something very similar to this. Our use-case is running some data crunching on very large instances types, and to minimise costs we would like the instances to terminate a soon as the job completes.

One difference in our use-case is that we don't want the terminated instance to be replaced immediately with a fresh one (unless there is a scheduled job).

Would you be willing to accommodate this in your PR?

I think the change would be, instead of marking the instance unhealthy, using [terminate-instance-in-auto-scaling-group](https://docs.aws.amazon.com/cli/latest/reference/autoscaling/terminate-instance-in-auto-scaling-group.html) with the `--should-decrement-desired-capacity` flag, which could be configurable.

Alternatively happy to submit this change myself as a new PR.

Reply:
@tekumara okay, I pushed up a potential implementation in 724d56c. Let me know if you think this would work for you.

Reply:
Fantastic, yes I think that will do the trick! Thankyou 🙏